internal/checkers/helpers_basic_type.go

```diff
@@ -56,9 +56,14 @@ func isTypedIntNumber(e ast.Expr, v int, types ...string) bool {
 	return false
 }
 
-func isIntNumber(e ast.Expr, v int) bool {
+func safeTypedBasicLit(e ast.Expr, typ token.Token) (*ast.BasicLit, bool) {
```

Comment:
I'm unsure about the name, but thanks I had also noticed this could be refactored into a dedicated function

Reply:
I'm so lame at wording. I'm only able to argue when I find a wording, but I'm never able to find a descent one to suggest 😅